api/applications/views.py

```diff
@@ -98,3 +105,31 @@ def perform_update(self, serializer):
         serializer.validated_data['owner'] = self.request.user
         # TODO: Write code to transfer ownership
         serializer.save(owner=self.request.user)
+
+
+class ApplicationReset(generics.RetrieveUpdateDestroyAPIView, ApplicationMixin):
```

Comment:
I'll let Matt answer to the final design choice. Two initial considerations were:

- Resetting is not something we want to be done lightly (it would force every single user of a service to re-authorize, which is a big headache)
- The user does not get to choose the new client secret- it's randomly generated. In that sense, it's not a normal request to update the data associated with an instance- it's an entirely separate kind of action. ("reset, but don't use any of the information I provide as part of the request")